corehq/apps/saved_reports/tasks.py

```diff
@@ -43,14 +45,45 @@ def send_delayed_report(report_id):
 def send_report(notification_id):
     notification = ReportNotification.get(notification_id)
 
+    # Intentionally let this raise an IndexError if there's no report.
+    # Shouldn't ever happen, and if it does, it means there's something wrong
+    # with our queuing or concurrency model.
+    record = ScheduledReportRecord.objects.filter(
+        state=ScheduledReportRecord.States.queued,
+        scheduled_report_id=notification_id,
+    ).order_by('-scheduled_for')[0]
+
+    # Mark all previous queued record as skipped
+    ScheduledReportRecord.objects.filter(
+        state=ScheduledReportRecord.States.queued,
+        scheduled_report_id=notification_id,
+        scheduled_for__lt=record.scheduled_for,
+    ).update(state=ScheduledReportRecord.States.skipped)
+
     # If the report's start date is later than today, return and do not send the email
     if notification.start_date and notification.start_date > datetime.today().date():
```

Comment:
could move this before the previous statement which would reduce the bulk updates by 1

Reply:
I suppose it would. Okay I'll do that.